extension/background/youtubei.js

```diff
@@ -54,3 +60,105 @@ function parseTimestamp(ts) {
     const hours = parseInt(parts[2]) || 0
     return secs + (60 * mins) + (60 * 60 * hours)
 }
+
+// time-comments related code below
+
+const INNERTUBE_API_KEY = "AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8"
+const INNERTUBE_CLIENT_NAME = "WEB"
+
+async function fetchTimeComments(videoResponse) {
+    const comments = await fetchComments(videoResponse)
+
+    // Currently using only the first minimally suitable comment.
+    // Maybe later implement more sophisticated comment selection.
+    for (let i = 0; i < comments.length; i++) {
+        const tsContexts = getTimestampContexts(comments[i].text)
+        if (tsContexts.length) {
+            return tsContexts
+        }
+    }    
+
+    return []
+}
+
+function getTimestampContexts(text) {
+    const TIMESTAMP_PATTERN = /^((?:\d?\d:)?(?:\d?\d:)\d\d)\s(.+)$/
+    const chapters = []
+    const lines = text.split("\r\n")
+
+    for (let i = 0; i < lines.length; i++) {
+        const tsMatch = lines[i].match(TIMESTAMP_PATTERN)
+        if (!tsMatch) {
+            return []
+        }
+
+        const timestamp = tsMatch[1]
+        const title = tsMatch[2]
+        const time = parseTimestamp(timestamp)
+
+        chapters.push({
+            title,
+            timestamp,
+            time,
+        })
+    }
+
+    return chapters
+}
+
+async function fetchComments(videoResponse) {
+    let token = commentsContinuationToken(videoResponse)
+    if (!token) {
+        return []
+    }
+    const comments = []
+    const commentsResponse = await fetchNext(token)
+
+    const items = commentsResponse.onResponseReceivedEndpoints[1].reloadContinuationItemsCommand.continuationItems
```

Comment:
Don't we need `appendContinuationItemsAction` as it's made in https://github.com/ris58h/youtube-timestamps/blob/master/extension/background/youtubei.js#L34-L35 ?

Reply:
When modifying the existing code, I took the result from the ternary expression branch for pageCount === 0. As I removed the while loop, pageCount cannot be greater than zero. If I got something wrong, please explain.